core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java

```diff
@@ -60,17 +61,9 @@ public void testSimpleEvictionMaxEntries() throws Exception {
          cache.put("key-" + (i + 1), "value-" + (i + 1));
       }
       assertEquals(CACHE_SIZE, cache.size());
-      int expectedEvictions;
-      if (getEvictionStrategy() == EvictionStrategy.LIRS) {
-         // Eviction count will be Size + (Size * .05) rounded up (since the first elements will be in resident blocks
-         // so they won't cause evictions to occur
-         expectedEvictions = (int) Math.ceil(CACHE_SIZE + (CACHE_SIZE * .05));
-      } else {
-         // Otherwise is LRU and that will evict on each write
-         expectedEvictions = CACHE_SIZE * 2;
-      }
-      assertEquals("eviction events count should be same with case size: " + evictionListener.getEvictedEvents(),
-            expectedEvictions, evictionListener.getEvictedEvents().size());
+      // We don't know for sure how many will be evicted due to randomness, but we know they MUST evict
+      // at least a size worth since we are writing more than double
+      assertTrue(evictionListener.evictedEntries.size() > CACHE_SIZE);
```

Comment:
The cache uses an admission window to monitor the new arrivals. When an insert pushes an entry out of this window into the next generation, that "candidate" entry and the eviction policy's victim are evaluated. Which ever has the largest frequency is retained. There are a few unpredictable aspects to this,
 - The frequencies are stored in a popularity sketch so collisions can raise the estimate
 - A "warm" candidate may be retained instead of a "hot" victim, randomly, to protect from HashDoS attacks.
 - A large "weighted" entry might cause the admission window to flush, but then itself not pass the filter and be removed. Imagine a cache size of 2MB with 100kb entries, and then the insertion of a 1.5MB entry. If that large entry has no historic frequency, it might be rejected when it forces itself to be promoted.